packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -167,6 +167,13 @@ class _CupertinoTabScaffoldState extends State<CupertinoTabScaffold> {
   @override
   void didUpdateWidget(CupertinoTabScaffold oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (_currentPage >= widget.tabBar.items.length) {
+      // Clip down to an acceptable range.
+      _currentPage = widget.tabBar.items.length - 1;
+      // Sanity check, since CupertinoTabBar.items's minimum length is 2.
+      assert(_currentPage >= 0);
```

Comment:
If this assert() is expected to blowup on a developer, can you add an appropriate explanation of the problem and a reasonable solution?

Reply:
Added a message here and in CupertinoTabBar